os/arch/arm/src/armv7-m/up_schedyield.c

```diff
@@ -0,0 +1,196 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <debug.h>
+#include <queue.h>
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+#include <stdint.h>
+#endif
+
+#include <tinyara/sched.h>
+#if CONFIG_RR_INTERVAL > 0
+#include <tinyara/clock.h>
+#endif
+#ifdef CONFIG_ARMV7M_MPU
+#include <tinyara/mpu.h>
+#endif
+#ifdef CONFIG_TASK_SCHED_HISTORY
+#include <tinyara/debug/sysdbg.h>
+#endif
+
+#include "sched/sched.h"
+#include "up_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+extern uint32_t g_umm_app_id;
+#endif
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_schedyield
+ *
+ * Description:
+ *   Called when current task is willing to
+ *   yield cpu resource to other ready-to-run task.
+ *   context switch would occur in below case:
+ *
+ *   1) The priority of the currently running task is either equal or more than other ready to run task
+ *
+ ****************************************************************************/
+
+void up_schedyield(void)
+{
+	struct tcb_s *rtcb = this_task();
+	struct tcb_s *ntcb = rtcb->flink;
+
+	if (ntcb == NULL) {
+		sdbg("Fail to yield, there is no next tcb. Current task : %d\n", rtcb->pid);
+		return ERROR;
+	}
```

Comment:
And I'm not sure we need ERROR return. Based on your implementation, if there is no tcb next to the current, yield will be failed. It means current tcb still has the context. 
But if we return OK, even though we will not make context switching, current tcb still has the context.
For context, it is same situation regardless of the return. So I think it is good to return OK for all case with dbg msg.

Reply:
agree. removed and return OK always.